edx/analytics/tasks/course_blocks.py

```diff
@@ -0,0 +1,501 @@
+from edx.analytics.tasks.decorators import workflow_entry_point
+from edx.analytics.tasks.course_list import TimestampPartitionMixin, CourseListPartitionTask
+
+
+log = logging.getLogger(__name__)
+
+
+class CourseBlockRecord(Record):
+    """
+    Represents a course block as fetched from the edX Course Blocks REST API, augmented with details about its
+    position in the course.
+    """
+    block_id = StringField(length=255, nullable=False, description='Block identifier.')
+    course_id = StringField(length=255, nullable=False, description='Identifier for the course containing the block.')
+    block_type = StringField(length=255, nullable=False, description='Block type.')
+    display_name = StringField(length=255, nullable=False, description='User-facing title of the block.')
+    is_root = BooleanField(default=False, nullable=False,
+                           description='True if the block is the course\'s root node.')
+    has_multiple_parents = BooleanField(default=False, nullable=False,
+                                        description='True if the block has more than one parent, making the course '
+                                                    'a directed acyclic graph.  If True, then parent_block_id and '
+                                                    'course_path will be null.')
+
+    parent_block_id = StringField(length=255, nullable=True,
+                                  description='Block identifier for the block\'s parent. If null, and '
+                                              'is_root and has_multiple_parents are False, then this block has been '
+                                              'removed from the course.')
+
+    course_path = StringField(nullable=True,
+                              description='Concatenated string of parent block display_name values, from '
+                                          'the root node to the parent_block_id.  Will be null if '
+                                          'has_multiple_parents.')
+    sort_idx = IntegerField(nullable=True,
+                            description='Number indicating the position that this block holds in a course-outline '
+                                        'sorted list of blocks.   Will be null if has_multiple_parents.')
+
+
+class CourseBlocksDownstreamMixin(TimestampPartitionMixin, WarehouseMixin, MapReduceJobTaskMixin):
+    """Common parameters used by the Course Blocks Data and Partition tasks."""
+
+    course_ids = luigi.Parameter(
+        description='List of course_id values to fetch course_block data for.  Accepts a list or JSON-formatted string '
+                    'defining an array of course_id strings.'
+    )
+    partition_format = luigi.Parameter(
+        config_path={'section': 'course_blocks', 'name': 'partition_format'},
+        default='%Y%m%d',
+        description="Format string for the course blocks table partition's `date` parameter.n"
```

Comment:
Nit: Consistently using the same quote style would be nice - to switch to single-quotes here would require some escaping, though, so it's not strictly necessary.